### PR TITLE
EVG-16040 Update test endpoint to return latest execution

### PR DESCRIPTION
**Pull request by bynn**
[EVG-16040](https://jira.mongodb.org/browse/EVG-16040)

### Description 
adding latest=true will give the most recent execution and default behavior is not modified
only works if execution is not specified 

### Testing 
staging
image

https://evergreen-staging.corp.mongodb.com/rest/v2/tasks/evg_race_detector_test_model_pod_4a1ed094bee959b235b34f3a2cffd71a43586e02_22_05_04_14_10_29/tests?latest=true

### CODE REVIEW

File: rest/route/test.go [hunk 1]
```diff
@@ -69,6 +69,8 @@ func (tgh *testGetHandler) Parse(ctx context.Context, r *http.Request) error {
 				StatusCode: http.StatusBadRequest,
 			}
 		}
+	} else {
+		tgh.testExecution = projCtx.Task.Execution
```
Comment (malikchaya2) [inline rest/route/test.go:73]: is projCtx.Task always the latest task?

Comment (bynn) [reply rest/route/test.go:73]: yeah i think all the older ones get archived so the one we do get should always be the latest execution

Comment (malikchaya2) [reply rest/route/test.go:73]: true

**Review by malikchaya2 — APPROVED** [review]

**Comment by bynn** [comment]
default option is now the same but you can specify latest 
cannot specify latest and also an execution

**Review by ablack12 — APPROVED** [review]
Remember to add to documentation and the ticket

**Comment by bynn** [comment]
https://github.com/evergreen-ci/evergreen/wiki/REST-V2-Usage#get-tests-from-a-task

**Pull request merged**